bob.clj

```diff
@@ -0,0 +1,31 @@
+(ns exercises.bob)
+
+(def whatever-phareses ["Tom-ay-to, tom-aaaah-to." 
+                        "Let's go make out behind the gym!"
+                        "It's OK if you don't want to go to the DMV."
+                        "Ending with ? means a question."
+                        "1, 2, 3"])
+
+(def woah-phareses ["WATCH OUT!"
+                   "WHAT THE HELL WERE YOU THINKING?"
+                    "ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!"
+                    "1, 2, 3 GO!"
+                    "I HATE YOU"])
+
+(def sure-phareses ["Does this cryogenic chamber make me look fat?"
+                   "4?"])
+
+(defn response-for
+  [phrase]
+  (cond
+    (some (partial = phrase) whatever-phareses)
+    "Whatever."
+    
+    (some (partial = phrase) woah-phareses)
```

Comment:
That is a clever use of partial to check inside a collection for a value. It is quite useful with lazy collections and so on.

There is an alternative to this implementation, where we could take advantage of the underlying datastructure to not be lazy.

This could be shortened using [contains?](https://clojuredocs.org/clojure.core/contains_q) to check if that collection contains the phrase.

One step further, we could optimize this code and use `#{}` (sets) when defining the phrases. We are not worried about ordering on this scenario, all phrases are unique and we only want to check for the presence. Sets are wonderful for that!

The combo of `contains?` and `set` would help to make this fast and concise.

Oh, and one tip about sets: they also implement the call protocol, so they are allowed to be used as functions:
They use the `get` call for that, which on most scenarios are like a `contains?` (unless you expect to check if the collection contains `nil`, on this scenario, only `(contains? x nil)` would work.)

The combo of `nil` and `false` being the only false-y values let you do things like this.

```clojure
(def a #{:a :b :c})

(when :a
  (println ":a"))


(when (get a :a)
  (println "get"))

(when (contains? a :a)
  (println "contains"))

(when (a :a)
  (println "set as function"))

(when (:a a)
  (println "keyword as function"))

```